lib/spack/spack/hooks/sbang.py

```diff
@@ -147,11 +151,39 @@ def install_sbang():
             spack.paths.sbang_script, sbang_path):
         return
 
-    # make $install_tree/bin and copy in a new version of sbang if needed
+    # remove outdated sbang if it exists
+    if os.path.exists(sbang_path):
+        os.remove(sbang_path)
```

Comment:
Downside here is that if we remove sbang first thing, then if anything goes wrong below we lose our `sbang`.  Can you modify this to copy and chmod a file, `.sbang.tmp`, alongside `sbang`, then move it into place *last*?

Reply:
I guess the idea is that you have to delete the file because you may not have permission to chmod it?

Reply:
@pwablito: the initial remove need to be deleted as well

Reply:
Ah, forgot to do that. Fixed in 3c31801

Reply:
Unfortunately this is what leads to races.